app/services/proxy_deployment_service.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class ProxyDeploymentService
+  delegate :apicast_configuration_driven,
+           :deployable?,
+           :service_mesh_integration?,
+           :provider,
+           :proxy_configs, to: :@proxy
+
+  class UnknownStageError < ArgumentError; end
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(proxy, environment: :staging, v1_compatible: false)
+    @proxy = proxy
+    @environment = environment
+    @v1_compatible = v1_compatible
+  end
+
+  def call
+    case @environment
+    when :staging, :sandbox
+      deploy
+    when :production
+      deploy_production
+    else
+      raise UnknownStageError, "Unknown environment: #{@stage}"
+    end
+  end
+
+  private
+
+  def deploy
+    return true unless deployable?
+
+    if service_mesh_integration?
+      deploy_v2 && deploy_production_v2
+    elsif apicast_configuration_driven
```

Comment:
```suggestion
    elsif apicast_configuration_driven?
```

Reply:
Is your suggestion to call [that method](https://github.com/3scale/porta/blob/3c20040cb487edc35e7a1b87e05fa89ee61d18f5/app/helpers/api/integrations_helper.rb#L158) instead or to rename this one?

Reply:
I think boolean methods are more explicit
Just for readability and convention

Reply:
I agree. Do you mean I create an alias? The method `apicast_configuration_driven` is already defined and I don't think it's a good idea to refactor it.

Reply:
Yes an alias is OK for me

by the way it is also in an helper https://github.com/3scale/porta/blob/b13c634f92c4b5c121528cfcd434a175bda8be4a/app/helpers/api/integrations_helper.rb#L158-L161

Reply:
I saw it but couldn't find a way to use it cleanly, I'd rather use the alias if that's OK.